### ISSUE TITLE
Better differentiation between 2,4 GHz and 5 GHz mode

### ISSUE THREAD

**Issue by fbruetting** [opnsense/core#1518]
I have just 5 GHz antennas on my PC Engines APU and therefore want to prevent usage of the 2,4 GHz spectrum. This is a bit cumbersome to me, since there are several settings available, which affect this, but can be configured to contradictory settings. So here are my critics:

• Standard: 802.11na
    – As far as I know, "na" is no standard. Can this be renamed to something more obvious, like "5 GHz (802.11n/a)"?
• Channel: (auto)
    – Selection list should be restricted to just the frequencies usable with the above chosen standard
• Minimum standard: 802.11n
    – 802.11g is also available, but makes no sense if I just want 5 GHz operation, should therefore also be filtered by the "standard" setting.
• On the dashboard, my wifi is declared as "autoselect mode 11b \<hostap\>" – how can this be possible if I selected the 802.11na standard? Is my router in 2,4 GHz mode, now…?!

**Comment by AdSchellevis** [comment]
timeout due to inactivity

**Comment by fbruetting** [comment]
So this is no bug anymore, because time has passed by…?

**Comment by fichtner** [comment]
No it was closed because nobody works on it and we have no benefit from having hundreds of pending or stale tickets that hide visibility of tickets we would like to progress.

Do you offer to work on it? It’s trivial to reopen this, but it should mean something. 😊

**Comment by fbruetting** [comment]
Sorry, I don’t know the internals and sadly have also no time (and many other constructions sites already).

But how about giving closed (but practically open) bug reports tags like "Bug", "Re-track in future" and maybe "Help wanted". Maintainers of other GitHub projects also use this to keep the open issues list small, but to also having a good opportunity to review deferred issues.

Examples:
https://github.com/adapta-project/adapta-gtk-theme/issues?q=is%3Aissue+is%3Aclosed
https://github.com/adapta-project/adapta-gtk-theme/issues/538

**Comment by fichtner** [comment]
We have these, but it's impossible to know how to tag it without looking. I can tell you we only read the values from the kernel/ifconfig so my best guess is to look into the FreeBSD source code, read problem reports over at https://bugs.freebsd.org/bugzilla/ to see what / if we can do ... also granted that we have the same hardware to reproduce (I don't).

While "na" is no standard it seems common enough to pass it as such. This seems to be preferred by Juniper Networking as well... https://www.juniper.net/documentation/en_US/junos-space-apps/network-director2.5/topics/concept/wireless-80211.html

The channels are read from ifconfig directly. It advertises what we can set, even if it advertises it wrong, that's programatically better than filtering the list on top as well.

I don't understand this one, can you explain:

> • Minimum standard: 802.11n
> – 802.11g is also available, but makes no sense if I just want 5 GHz operation, should therefore also be filtered by the "standard" setting.

It would help to break questions out of reports and / or report them individually for clarity and the chances are higher that someone will jump right in to help / discuss.


Thanks,
Franco

**Comment by fbruetting** [comment]
Ok, I try to look into it the next days, seems not to be so difficult then.

My question (I hope I recall it correctly, since this is some time ago and I haven’t used OPNsens since):
I first specified 802.11na, what means 5 GHz operation only (if I understood it correctly). But afterwards I’m able to chose 802.11g which operates at 2,4 GHz. That makes no sense to me, since I selected 802.11na above – therefore, the “minimum standard” should be filtered appropriately.
